### ISSUE TITLE
Warnings about strings in eslintrc

### ISSUE THREAD

**Issue by ralfstx** [Microsoft/vscode-eslint#72]
The ESLint documentation suggests that the strings "error", "warn", and "off" can be used instead of the integer values, for [example](http://eslint.org/docs/rules/semi#always) `semi: ["error", "always"]`.

Currently, VSCode brings up a warning when using this syntax:

> Incorrect type. Expected one of integer, array

**Comment by mitorez** [comment]
+1 - I'm seeing the same issue. If it matters, I'm running:
-Windows 8.1 Enterprise 64-bit
-Visual Studio Code 1.1.1
-ESLint Extension 0.10.15
-Project ESLint 2.10.2
-Project Node 5.10.1
-Project npm 3.8.3

**Comment by dbaeumer** [comment]
The extension uses (hot) the schema from http://json.schemastore.org/eslintrc to validate the .eslintrc.json file. I looked at the file and the schema still requires 0, 1, 2. 

I tested this in the upcoming version of code and interestingly it doesn't complain anymore. Need to see hay.

**Comment by dbaeumer** [comment]
To fix this it is best to provide a PR against https://github.com/SchemaStore/schemastore.

**Comment by dbaeumer** [comment]
OK. Here is what I found out using eslint version 2.10.2 (the latest)

- using "off", "warning", "error" is not allowed of a rules if this is the primary property
```json
		"semi": "warning"
```
doing so result in the following error when executing eslint from the command line:
```
P:\mseng\VSCode\Playgrounds\linters\.eslintrc.json:
        Configuration for rule "semi" is invalid:
        Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"warning"').

Error: P:\mseng\VSCode\Playgrounds\linters\.eslintrc.json:
        Configuration for rule "semi" is invalid:
        Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed '"warning"').

    at validateRuleOptions (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config\config-validator.js:116:15)
    at P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config\config-validator.js:163:13
    at Array.forEach (native)
    at Object.validate (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config\config-validator.js:162:35)
    at Object.load (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config\config-file.js:518:19)
    at loadConfig (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config.js:64:33)
    at getLocalConfig (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config.js:126:23)
    at Config.getConfig (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\config.js:227:22)
    at processText (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\cli-engine.js:217:27)
    at processFile (P:\mseng\VSCode\Playgrounds\linters\node_modules\eslint\lib\cli-engine.js:292:18)
```
Using this inside an array works as expected and Code doesn't print any warning/error.
```json
		"semi": [
			"error",
			"always"
		]
```

I will close the issue since there is nothing VS Code can do about this.

**Comment by dbaeumer** [comment]
Pleaser ping if I misunderstood something.

**Comment by mitorez** [comment]
"warning" is not valid string value for ESLint, "warn" (1) is valid: http://eslint.org/docs/user-guide/configuring#configuring-rules

Adding [] around the string value does prevent VS Code messages, but it prevents it for invalid string values too (like "warning").

**Comment by dbaeumer** [comment]
Sorry for this. Agree that "warn" is accepted. So we need to fix this in the schema store.

**Comment by dbaeumer** [comment]
See https://github.com/SchemaStore/schemastore/pull/172

**Comment by dbaeumer** [comment]
I will close this issue since there is nothing else eslint can do. As soon as the PR is accepted all eslint plugins will get the new schema automatically since it is referenced as: http://json.schemastore.org/eslintrc

**Comment by mitorez** [comment]
@dbaeumer thank you
